tools/build-tools/src/buildVersion/buildVersion.ts

```diff
@@ -90,8 +91,60 @@ export function getSimpleVersion(file_version: string, arg_build_num: string, ar
     return fullVersion;
 }
 
+type TagPrefix = string | undefined | "client" | "server";
+
+/**
+ * Extracts versions from the output of `git tag -l` in the working directory. The returned array will be sorted
+ * ascending by semver version rules.
+ *
+ * @param prefix - the tag prefix to filter the tags by (client, server, etc.).
+ * @returns an array of versions extracted from the output of `git tag -l`.
+ */
+function getVersions(prefix: TagPrefix) {
+    const raw_tags = child_process.execSync(`git tag -l`, { encoding: "utf8" });
+    const tags = raw_tags.split(/\s+/g).map(t => t.trim());
+    const filtered = tags.filter(v => v.startsWith(`${prefix}_v`));
```

Comment:
It might be better to filter in getVersionsFromStrings so it doesn't need to make the assumption that the tags that got passed in is already filtered.

Reply:
I mean you don't need to filter in `getVersions` if you do it in `getVersionsFromStrings` (which you have added now.

Reply:
Fixed in latest.